packages/frontend/src/features/sqlRunner/components/visualizations/BarChart.tsx

```diff
@@ -27,10 +27,10 @@ const BarChart: FC<BarChartProps> = memo(
         return (
             <>
                 {!spec && <Skeleton h="100%" />}
+                <LoadingOverlay visible={loading} />
```

Comment:
```suggestion
                <LoadingOverlay visible={loading || !spec} />
```
since `error` is handled above ^

Reply:
and we can remove the `Skeleton` ^